### PR TITLE
reader: accumulate all errors from a file

### PR DESCRIPTION
**Pull request by adamdecaf**
Fixes: https://github.com/moov-io/ach/issues/265

### CODE REVIEW

File: reader_test.go [hunk 1]
```diff
@@ -317,14 +385,16 @@ func testFileFileHeaderErr(t testing.TB) {
...
 	r.File.Control = mockFileControl()
 	_, err := r.Read()
-	if p, ok := err.(*ParseError); ok {
-		if e, ok := p.Err.(*FieldError); ok {
-			if e.Msg != msgFieldInclusion {
-				t.Errorf("%T: %s", e, e)
+	if el, ok := err.(ErrorList); ok {
+		if p, ok := el.Err().(*ParseError); ok {
+			if e, ok := p.Err.(*FieldError); ok {
+				if e.Msg != msgFieldInclusion {
```
Comment (adamdecaf) [inline reader_test.go:423]: Asking clients to inspect like this is annoying, but will clients need to?

File: reader.go [hunk 2]
```diff
@@ -19,33 +21,104 @@ type ParseError struct {
...
+			fmt.Fprintf(w, "\n")
+		}
+	}
+}
+
+func (e ErrorList) Empty() bool {
+	return e == nil || len(e) == 0
+}
+
+func (e ErrorList) MarshalJSON() ([]byte, error) {
```
Comment (adamdecaf) [inline reader.go:83]: Should we render this in the endpoints as an array of errors? JSON arrays are ordered (per the spec), but I've seen clients not respect that.

**Comment by adamdecaf** [comment]
What happened to our code coverage testing?

File: reader_test.go (hunk 1 above)
Comment (bkmoovio) [reply reader_test.go:423]: I think clients would probably just output or log the error message.

File: reader.go (hunk 2 above)
Comment (bkmoovio) [reply reader.go:83]: I think so, Wade?  If so should we document that we are following the spec?

Comment (adamdecaf) [reply reader.go:83]: I'm tempted to just return a string (that has no format). I'd rather not go down the rabbit trail of string parsing error messages.

Comment (bkmoovio) [reply reader.go:83]: Are you saying you would return a string which contained all errors? If so,  how would they be distinguished for a client?

Comment (adamdecaf) [reply reader.go:83]: Right now it's a big string, with newlines and an indent.

```
first error
  second error
  ...
```

Comment (adamdecaf) [reply reader.go:83]: There's a test showing the format too: https://github.com/moov-io/ach/pull/299/files#diff-f86b763600291cbceee077a33133434aR53

Comment (wadearnold) [reply reader.go:83]: The spec is a suggestion on an implementation. We are free to change the specification however we see fit. I agree with @bkmoovio that the error will probably just be logged.

**Comment by adamdecaf** [comment]
https://moov-io.slack.com/archives/CAH29P84F/p1539885027000100 Approved in slack

**Pull request merged**